examples/networking/coap/nanocoap_server/main.c

```diff
@@ -43,15 +47,35 @@ int main(void)
         setup_observe_event();
     }
 
+#ifdef MODULE_LWIP_IPV4
+#define _TEST_ADDR4_LOCAL  (0x9664a8c0U)   /* 192.168.100.150 */
+#define _TEST_ADDR4_MASK   (0x00ffffffU)   /* 255.255.255.0 */
+
+    sys_lock_tcpip_core();
+    struct netif *iface = netif_find("ET0aa");
+    ip4_addr_t ip, subnet;
+    ip.addr = _TEST_ADDR4_LOCAL;
+    subnet.addr = _TEST_ADDR4_MASK;
+    netif_set_addr(iface, &ip, &subnet, NULL);
+    sys_unlock_tcpip_core();
+
+    printf("{\"IPv4 addresses\": [\"192.168.100.150\"]}\n");
```

Comment:
We don't want to set this unconditionally - LWIP does support DHCP, let's just use that.

Reply:
Yes, I know that LWIP support DHCP. But this required additional DHCP server, and this complicates setup.

Do you mean that this code shoul choose to use DHCP or static address?

Reply:
Whether DHCP makes things more complicated depends on the setup: On a real board that you run in your home network, DHCP makes things a lot easier, a static address in a different subnet will not work.

And even on `native` you'll need some configuration on the host side to route 192.168.100.150 to your tap interface.